dali/ft/filecopy.cpp

```diff
@@ -3224,8 +3224,70 @@ void FileSprayer::updateTargetProperties()
             }
 
             // Keep source kind
-            curProps.setProp("@kind", srcAttr->queryProp("@kind"));
+            if (srcAttr->hasProp("@kind"))
+                curProps.setProp("@kind", srcAttr->queryProp("@kind"));
```

Comment:
should you also check srcAttr->hasProp("@format") in similar way?

Reply:
@ghalliday - is it possible for source and target to be different kinds? e.g. spray csv to flat file?

Reply:
@ghalliday is there any conclusion related this question?

Reply:
It is possible for the source to be utf16 and the target to be utf8.  There are situations where files are sprayed as csv, but actually ready as xml, but I don't think the basic csv/xml/flat format ever changes.

Reply:
ok, then I think that possibility is not an issue for this JIRA/PR - perhaps it _should_ be possible to spray and change target published format, but not something that's currently possible.

Reply:
> https://github.com/hpcc-systems/HPCC-Platform/pull/12380/files#r278079968

@AttilaVamos  Not sure if this has/needs to be addressed ?

i.e. shouldn't above code look something like:
```
            if (srcAttr->hasProp("@kind"))
            {
                curProps.setProp("@kind", srcAttr->queryProp("@kind"));
                if (srcAttr->hasProp(FPformat))
                    curProps.setProp(FPformat, srcAttr->queryProp(FPformat));
            }
```